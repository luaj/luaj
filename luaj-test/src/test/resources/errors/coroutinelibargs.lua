package.path = "?.lua;test/lua/errors/?.lua"
require 'args'

-- arg type tests for coroutine library functions

-- coroutine.create
banner('coroutine.create')
checkallpass('coroutine.create',{somefunction})
checkallerrors('coroutine.create',{notafunction},'bad argument')

-- coroutine.resume
banner('coroutine.resume')
local co = coroutine.create(function() while true do coroutine.yield() end end)
checkallpass('coroutine.resume',{{co},anylua})
checkallerrors('coroutine.resume',{notathread},'bad argument')

-- coroutine.running
banner('coroutine.running')
checkallpass('coroutine.running',{anylua})

-- coroutine.status
banner('coroutine.status')
checkallpass('coroutine.status',{{co}})
checkallerrors('coroutine.status',{notathread},'bad argument')

-- coroutine.wrap
banner('coroutine.wrap')
checkallpass('coroutine.wrap',{somefunction})
checkallerrors('coroutine.wrap',{notafunction},'bad argument')

-- coroutine.yield
banner('coroutine.yield')
local function f() 
	print( 'yield', coroutine.yield() )
	print( 'yield', coroutine.yield(astring,anumber,aboolean) )
	error('error within coroutine thread')
end
local co = coroutine.create( f )
print( 'status', coroutine.status(co) )
print( coroutine.resume(co,astring,anumber) )
print( 'status', coroutine.status(co) )
print( coroutine.resume(co,astring,anumber) )
print( 'status', coroutine.status(co) )
local s,e = coroutine.resume(co,astring,anumber)
print( s, string.match(e,'error within coroutine thread') or 'bad message: '..tostring(e) )
print( 'status', coroutine.status(co) )

