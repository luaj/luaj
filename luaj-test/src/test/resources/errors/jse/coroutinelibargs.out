====== coroutine.create ======
--- checkallpass
-		coroutine.create(<function>)	<thread>
--- checkallerrors
-		coroutine.create(nil)	...bad argument...
-		coroutine.create('abc')	...bad argument...
-		coroutine.create(1.25)	...bad argument...
-		coroutine.create(true)	...bad argument...
-		coroutine.create(<table>)	...bad argument...
-		coroutine.create(<thread>)	...bad argument...
====== coroutine.resume ======
--- checkallpass
-		coroutine.resume(<thread>,nil)	true
-		coroutine.resume(<thread>,'abc')	true
-		coroutine.resume(<thread>,1.25)	true
-		coroutine.resume(<thread>,true)	true
-		coroutine.resume(<thread>,<table>)	true
-		coroutine.resume(<thread>,<function>)	true
-		coroutine.resume(<thread>,<thread>)	true
--- checkallerrors
-		coroutine.resume(nil)	...bad argument...
-		coroutine.resume('abc')	...bad argument...
-		coroutine.resume(1.25)	...bad argument...
-		coroutine.resume(true)	...bad argument...
-		coroutine.resume(<table>)	...bad argument...
-		coroutine.resume(<function>)	...bad argument...
====== coroutine.running ======
--- checkallpass
-		coroutine.running(nil)	<thread>,true
-		coroutine.running('abc')	<thread>,true
-		coroutine.running(1.25)	<thread>,true
-		coroutine.running(true)	<thread>,true
-		coroutine.running(<table>)	<thread>,true
-		coroutine.running(<function>)	<thread>,true
-		coroutine.running(<thread>)	<thread>,true
====== coroutine.status ======
--- checkallpass
-		coroutine.status(<thread>)	'suspended'
--- checkallerrors
-		coroutine.status(nil)	...bad argument...
-		coroutine.status('abc')	...bad argument...
-		coroutine.status(1.25)	...bad argument...
-		coroutine.status(true)	...bad argument...
-		coroutine.status(<table>)	...bad argument...
-		coroutine.status(<function>)	...bad argument...
====== coroutine.wrap ======
--- checkallpass
-		coroutine.wrap(<function>)	<function>
--- checkallerrors
-		coroutine.wrap(nil)	...bad argument...
-		coroutine.wrap('abc')	...bad argument...
-		coroutine.wrap(1.25)	...bad argument...
-		coroutine.wrap(true)	...bad argument...
-		coroutine.wrap(<table>)	...bad argument...
-		coroutine.wrap(<thread>)	...bad argument...
====== coroutine.yield ======
status	suspended
true
status	suspended
yield	abc	1.25
true	abc	1.25	true
status	suspended
yield	abc	1.25
false	error within coroutine thread
status	dead
